runelite-client/src/main/java/net/runelite/client/plugins/microbot/tutorialisland/TutorialIslandScript.java

```diff
@@ -337,8 +389,11 @@ public void MageGuide() {
 
         if (Microbot.getVarbitPlayerValue(281) == 610 || Microbot.getVarbitPlayerValue(281) == 620) {
             WorldPoint worldPoint = new WorldPoint(3141, 3088, 0);
-            if (Rs2Player.distanceTo(worldPoint) >= 4) {
-                Rs2Walker.walkTo(worldPoint, 4);
+            WorldPoint targetPoint = (npc != null) ? npc.getWorldLocation() : worldPoint;
+            int distance = Rs2Player.distanceTo(targetPoint);
+
+            if (distance >= 8) {
```

Comment:
this has to be > 8, otherwise it can get stuck in an infinite loop, because the walker will not walk if it is equal to 8